lib/Scheduler/sched_basic_data.cpp

```diff
@@ -350,7 +350,8 @@ SchedInstruction *SchedInstruction::GetNxtPrdcsr(InstCount *scsrNum,
 
 SchedInstruction *SchedInstruction::GetFrstScsr(InstCount *prdcsrNum,
                                                 UDT_GLABEL *ltncy,
-                                                DependenceType *depType) {
+                                                DependenceType *depType,
+                                                bool *IsArtificial) {
```

Comment:
Pass by reference?

Reply:
I copied the way it was done to retrieve an optional value from a GraphEdge. Is there an issue with this? Although I do agree we can definitely use a better method to retrieve these values.